Algorithm/QCAlgorithm.cs

```diff
@@ -1809,5 +1810,21 @@ public void SetStatus(AlgorithmStatus status)
         {
             Status = status;
         }
+
+        /// <summary>
+        /// Downloads the requested resource as a <see cref="string"/>.
+        /// The resource to download is specified as a <see cref="string"/> containing the URI.
+        /// </summary>
+        /// <param name="address">A string containing the URI to download</param>
+        /// <param name="userName">The user name associated with the credentials</param>
+        /// <param name="password">The password for the user name associated with the credentials</param>
+        /// <returns>The requested resource as a <see cref="string"/></returns>
+        public string Fetch(string address, string userName = null, string password = null)
+        {
+            using (var client = new WebClient { Credentials = new NetworkCredential(userName, password) })
+            {
+                return client.DownloadString(address);
```

Comment:
Also, we might want to setup the proxy as we do similarly in the `RemoteFileSubscriptionStreamReader` so it will use the local machine's proxy settings. This was a PR from a few years ago and helps people when they run locally through a proxy.